### DISCUSSION TITLE
同一个job中相同source，cdc为什么要多次同步

### DISCUSSION THREAD

**Discussion by lykpass** [apache/flink-cdc#1042] [Q&A] [answered]
![pic](https://user-images.githubusercontent.com/48595535/161243082-5ea222cc-f281-43af-84cc-8a6e7b47a71e.jpg)
我在一个job中执行了两次sqlQuery都用到了basics_warehouse这张表，cdc为什么要同步两次呢

**Comment by shadon178** [comment, marked as answer]
你最好把代码细节也弄出来，不然没法定位问题的。

**Reply by lykpass** [reply]
就是一个任务中  我执行了两个 sqlQuery，这两个sqlQuery中都用到了这个表，然后这个表作为source就会同步两次

**Reply by lykpass** [reply]
> 你最好把代码细节也弄出来，不然没法定位问题的。

通过 sqlQuery createTemporaryView   注册视图 crm_org_info_test0
	
逻辑如下	：

select 
	id,node_name,is_leaf
from 
	crm_org_info
where 
	id > 10


另外两个sqlQuery查询依赖视图	crm_org_info_test0

如下：

  sqlQuery：
	  
  select 
	  id,node_name,is_leaf
  from 
	  crm_org_info_test0
  where 
	  id > 10000


  sqlQuery：
	  
  select 
	  id,node_name
  from 
	  crm_org_info_test0
  where 
	  is_leaf = 1


但是生成的DAG图中显示crm_org_info这个表被同步了两次数据，是不是可以同步一次达到效果的


![image](https://user-images.githubusercontent.com/48595535/174048515-bb1f711b-2a52-4a11-806c-0c273258a1fa.png)
